### ISSUE TITLE
GoInstall clears my editing window when errors exist

### ISSUE THREAD

**Issue by dune-rocks** [fatih/vim-go#683]
When I run `:GoInstall` with the following program, the window containing this code gets cleared (there's zero lines in it) and the just the small window at the bottom with the error remains. 

```go
import (
	"fmt"
	"os"
	"strings"
)

func main() {
	paths := os.Getenv("PATH")
	paths = strings.Split(paths, string(os.PathListSeparator))
	for _, p := range paths {
		fmt.Println(p)
	}
}
```

Does anyone else notice this with their installation? I'm a newbie, so it's likely I've screwed something up, but the behaviour smells buggy. After this happens, I have to reopen the file in Vim.

**Comment by Rmanolis** [comment]
This happens to me also

**Comment by fatih** [comment]
Are you using the latest master? It was fixed 1-2 weeks ago.

**Comment by Rmanolis** [comment]
Yes I am sure, I just updated the plugin and the error continues . 
I use Vim 7.4.1046 modified by pi-rho

**Comment by fatih** [comment]
Ok please provide a detailed test case. An example code, the directory you're calling it with exact commands, your vimrc, etc... This is important so I can reproduce it. Thanks!

**Comment by Rmanolis** [comment]
My vmirc (I am shy because it is not the cleanest)
https://gist.github.com/Rmanolis/8a8d9d252fe7f09395f5
I use golang 1.5.3
My project folders are 
ctrls  utils  models  requirements.sh  gin-bin  main.go  confs  webdev  static  cbcli  TODO.md
where in the ctrls has an auth.go
https://gist.github.com/Rmanolis/38a85a441096b483296b
There when I make any error and press ":GoInstall", it opens a new buffer with the same name of the file but on the main folder and not in the "ctrls". 
I dont know what else would you need. 
However I am going to sleep, if anything ask me and I will answer tomorow. 
Have a nice day!

**Comment by fatih** [comment]
Thanks @Rmanolis but your example is not something I can use. Provide a simple `main.go` which I can use to reproduce the problem. The information you give me is not useful! Thanks.

**Comment by Rmanolis** [comment]
sorry that I didnt reply earlier, I got very sick. 
However with the new plugin it is working correctly. 
Thanks for you hard work :)

**Comment by fatih** [comment]
This is now fixed with #692
